Blog post on .Netcore C# client and query (#269)
infinispan/infinispan.github.io: _posts/2022-01-18-dotnet-core-query.adoc

---
layout: blog
title: Querying Infinispan with the new .NET core client
permalink: /blog/:year/:month/:day/dotnet-core-query

> date: '2022-01-18T12:00:00.000-00:00'

Reviewer: ```suggestion
date: '2022-01-21T12:00:00.000-00:00'
```

author: rigazilla
tags: [ "hotrod", ".Net Core", "client", "C#", "query", "query" ]
---

Dear Infinispanners and .NET Core lovers,

> time is passed, keys have been pressed and features have been implemented since

Reviewer: ```suggestion
Time has passed, keys have been pressed and features have been implemented since
```

we https://infinispan.org/blog/2021/06/21/infinispan-dotnet-core-client[presented],
some months ago, the new .NET core client project.

> To show the progresses done on the project we decided to write a demo/tutorial on

Reviewer: ```suggestion
To show you our progress with the project we've decided to write a demo/tutorial on
```

> how you can run remote queries against Infinispan in your .NET Core application.

Reviewer: ```suggestion
how you can run remote queries against Infinispan Server deployments in your .NET Core application.
```

> The demo aims to proof the overall maturity of the client and highlight some key client features like:

Reviewer: ```suggestion
The demo aims to prove the overall maturity of the client and highlight some key client features like:
```

> * it's shipped as a nuget package, so it can be easily included in a C# project;

Reviewer: ```suggestion
* Shipping as a NuGet package, making it easy to include in C# projects.
```

> * it's based on non blocking core, thus it has a native support for concurrency;

Reviewer: ```suggestion
* Using Infinispan's non-blocking core to add native support for concurrency.
```

Author: @oraNod, I meant the client uses BeetleX non-blocking library as its core. Maybe we could say:
` * Using [BeetleX's](https://github.com/beetlex-io/BeetleX) non-blocking core to add native support for concurrency.

Reviewer: @rigazilla Thanks I updated the suggestion with that.

> * it supports language neutral data types with Protobuf definition.

Reviewer: ```suggestion
* Using language-neutral data types with the Protobuf media type for interoperability.
```

Want to know more go to the https://github.com/infinispan/Infinispan.Hotrod.Core/tree/query-demo[demo page] and try it!

> We hope this is a good 2022 start for the all Infinispan C# developers!

Reviewer: ```suggestion
We hope this is a good start to 2022 for all Infinispan C# developers!
```

> Please let us know your thoughts, a good place for them is the https://github.com/infinispan/Infinispan.Hotrod.Core/issues[issues] page.

Reviewer: ```suggestion
Please let us know your thoughts. A good place to share them is the https://github.com/infinispan/Infinispan.Hotrod.Core/issues[issues] page.
```
